### PR TITLE
77125k add ineligible medicare upload

### PR DESCRIPTION
**Pull request by michaelclement**
## Summary

Adds a file upload if the user is over 65 and ineligible for medicare.

- Added custom page to upload ineligibility doc
- Refactored page that collects user's Medicare enrollment status
- Updated existing unit tests

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/77125

## Testing done

- Manual testing
- Updated and ran existing tests to verify they still work

## Screenshots

|         | S1 | S2 |
| ------- | ------ | ----- |
| Desktop |updated_wording|upload_file|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

NA

### CODE REVIEW

File: src/applications/ivc-champva/10-10D/config/form.js [hunk 1]
```diff
@@ -699,7 +701,7 @@ const formConfig = {
           }),
         },
         page14: {
-          path: 'applicant-information/:index/ssn-dob',
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/config/form.js:702]: This screen doesn't collect DOB anymore.

File: src/applications/ivc-champva/10-10D/config/form.js [hunk 2]
```diff
@@ -1336,7 +1338,13 @@ const formConfig = {
...
           CustomPageReview: ApplicantMedicareStatusReviewPage,
           schema: applicantListSchema([], {
-            applicantMedicareStatus: { type: 'string' },
+            applicantMedicareStatus: {
+              type: 'object',
+              properties: {
+                eligibility: { type: 'string' },
+                otherIneligible: { type: 'string' },
+              },
+            },
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/config/form.js:1347]: Updating so it reflects the reality of the custom page's input. Also updated pages that check this via `depends` so they drill down the right property.

File: src/applications/ivc-champva/10-10D/tests/pages/applicantMedicareStatusPage.unit.spec.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ import mockData from '../fixtures/data/test-data.json';
 
 testComponentRender(
   'ApplicantMedicareStatusReviewPage',
-  <>{ApplicantMedicareStatusReviewPage()}</>,
+  <ApplicantMedicareStatusReviewPage data={mockData.data} title={() => ''} />,
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/tests/pages/applicantMedicareStatusPage.unit.spec.js:19]: Fixing this unit test now that the custom page is slightly different underneath.

**Review by balexandr — APPROVED** [review]
lgtm

**Review by cloudmagic80 — APPROVED** [review]

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**